apps/meteor/lib/utils/parseReaction.ts

```diff
@@ -0,0 +1,14 @@
+import type { RoomType } from '@rocket.chat/apps-engine/definition/rooms';
+
+import { getEmojiClassNameAndDataTitle } from '../../client/lib/utils/renderEmoji';
+
+export function parseReaction(reactionText: string, roomtype: RoomType): string {
+	let reaction = '';
+	if (roomtype === 'd') {
+		reaction = `reacted with ${getEmojiClassNameAndDataTitle(reactionText.trim()).children}`;
```

Comment:
what if we just use the reaction name and build the message on the clients?

anyway the translation is missing

Reply:
Hi @ggazzo ,
I have been working on a translation task to obtain reaction names from the back-end and build messages on the client-side. To achieve this, changes need to be made to one of the following files:

[konchatNotifications.ts](https://github.com/RocketChat/Rocket.Chat/blob/develop/apps/meteor/client/startup/notifications/konchatNotifications.ts) at line 86, where we receive the notification payload from the back-end.
[KonchatNotification.ts](https://github.com/RocketChat/Rocket.Chat/blob/develop/apps/meteor/app/ui/client/lib/KonchatNotification.ts) at line 36, where the above file calls this function.

However, since useTranslation is a hook and cannot be used in either of these files as they lack React components, I devised an alternative approach. Though it is somewhat verbose, it successfully translates the messages. I will make a commit with this working approach , please take a look.


<img width="334" alt="Screenshot 2024-06-22 at 10 30 22 PM" src="https://github.com/RocketChat/Rocket.Chat/assets/83659881/dd1e51e2-6bb1-4ad9-a735-32a0e517b885">